docs/self-hosted/kubernetes-troubleshooting.md

```diff
@@ -36,3 +36,25 @@ System.
 > a year. As Private Packagist only allows a time-drift of up to one (1) minute, we
 > recommend using TOTP devices that have the ability to stay synchronized with
 > the correct time (such as a phone, or re-programmable TOTP hardware devices).
+
+#### Issues with Reverse-Proxy running in front of the Kubernetes Cluster
+
+Please follow the instructions below, if you are experiencing problems with the reverse-proxy not being able to connect to
+the cluster and encountering errors like this:
+```
+Peer closed connection in SSL handshake (104: Connection reset by peer) while SSL handshaking to upstream
+```
+
+Ensure that the SNI (Server Name Indication) TLS Extension is properly set for requests to the ingress controller of the Kubernetes Cluster.
+This is not the case when using IPs in `proxy_pass` on the reverse proxy and will result in an SSL handshake error.
```

Comment:
@pscheit Also pinging you to get your opinion here :)

Reply:
Note, alternatively we just move the last sentence to here and mention that the "following examples assume you are using nginx, please consult the documentation of other reverse-proxy servers to achieve the same result.", something like that. Just something to avoid confusion :)

Reply:
the last suggestion is better, because now the two sentences basically tell us the same story?